src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
```

Comment:
this can throw a `NullPointerException` if 1throwableClass` is null. I recommend adding this to the top of the metod:

    if (throwableClass == null) {
      throw new NullPointerException("throwableClass"); // or some better message
    }

(yes, some might argue that `IllegalArgumentException` would be better, but JUnit has mostly settled on `NullPointerException` for new code (because that's what would usually be thrown if the check wasn't there), so we should be consistent.